js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,681 @@
+            'node_modules/foo/package.json': manifestBuilder(
+              exportsBuilder(key)
+            ),
+          });
+          dir.chdir(
+            () => {
+              const result = compileString('@use "pkg:foo/colors";', {
+                importers: [nodePackageImporter],
+              });
+              expect(result.css).toEqualIgnoringWhitespace('a {b: c;}');
+            },
+            {changeEntryPoint: 'index.js'}
+          );
+        }));
+    });
+    it('compiles with first conditional match found', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+          'node_modules/foo/src/sass/_variables.scss':
+            'a {from: sassCondition}',
+          'node_modules/foo/package.json': manifestBuilder({
+            exports: {
+              '.': {
+                sass: './src/sass/_variables.scss',
+                style: './src/sass/_styles.scss',
+              },
+            },
+          }),
+        });
+        dir.chdir(
+          () => {
+            expect(
+              compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              }).css
+            ).toEqualIgnoringWhitespace('a {from: sassCondition;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('throws if multiple exported paths match', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+          'node_modules/foo/src/sass/_variables.scss': 'a {b: c}',
+          'node_modules/foo/package.json': manifestBuilder({
+            exports: {
+              './index.scss': {sass: './src/sass/_variables.scss'},
+              './_index.sass': {sass: './src/sass/_styles.scss'},
+            },
+          }),
+        });
+        dir.chdir(
+          () => {
+            expect(() =>
+              compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              })
+            ).toThrowSassException({
+              includes: 'multiple potential resolutions',
+            });
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    it('resolves string export', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+          'node_modules/foo/src/sass/_variables.scss': 'a {b: c}',
+          'node_modules/foo/package.json': manifestBuilder({
+            exports: './src/sass/_variables.scss',
+          }),
+        });
+        dir.chdir(
+          () => {
+            expect(
+              compileString('@use "pkg:foo";', {
+                importers: [nodePackageImporter],
+              }).css
+            ).toEqualIgnoringWhitespace('a {b: c;}');
+          },
+          {changeEntryPoint: 'index.js'}
+        );
+      }));
+    describe('wildcards', () => {
```

Comment:
Probably also worth testing that something like `exports: {'./*': {sass: './src/sass/*'}}` works.